### PR TITLE
Add warning to private space creation flow

### LINKED ISSUE #3485: Add warning to private space creation flow
#3336 https://github.com/matrix-org/matrix-react-sdk/pull/6155

image
https://www.figma.com/file/baCEQP0BvtuitBJUAtapoc/Spaces-Beta?node-id=4641%3A111996

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3485
image

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt [hunk 1]
```diff
@@ -51,18 +53,38 @@ class SpaceCreationActivity : SimpleFragmentActivity(), CreateSpaceViewModel.Fac
...
+
+        override fun onFragmentDetached(fm: FragmentManager, f: Fragment) {
+            when (f) {
+                is BetaWarningBottomSheet -> {
+                    f.interactionListener = null
+                }
+            }
+            super.onFragmentDetached(fm, f)
+        }
+    }
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt:74]: We should find a way to improve this code, or at least do not duplicate it everywhere...

File: vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    var interactionListener: InteractionListener? = null
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?) =
+            BottomSheetSpaceCreatePrivateWarningBinding.inflate(inflater, container, false)
+
+    override val showExpanded = true
+
+    override fun onDetach() {
+        interactionListener = null
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt:40]: setting to null is done several times then?

**Review by bmarty — APPROVED** [review]
Some small remark, else LGTM

File: vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt:40]: yes, it's defensive coding, but can remove

File: vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt:74]: I changed it to use only inter-fragment communication without going through the hosting activity. 
Looks cleaner as only the two communicating fragments are modified
It's using #setFragmentResultListener and #setFragmentResult

File: vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/create/BetaWarningBottomSheet.kt:40]: see previous reply

**Review by bmarty — APPROVED** [review]
Thanks for the update. I will update the TextView style after the merge

**Pull request merged**